File: converted/utilities/is-plain-object/elide-is-plain-object.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-is-plain-object

This demonstrates calling the TypeScript is-plain-object implementation from Python
using Elide's polyglot capabilities.

Benefits:
- One is-plain-object implementation shared across TypeScript and Python
- Consistent object checking behavior across services
- No Python library inconsistencies
- Perfect for data pipelines and processing
"""

# NOTE: Exact syntax depends on Elide's Python polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Python support is ready

# Assuming Elide provides something like:
# from elide import require
# is-plain-object_module = require('./elide-is-plain-object.ts')

print("=== Python Calling TypeScript IsPlainObject ===\n")

# Example 1: Basic Usage
# result = is-plain-object_module.default(input_data)
# print(f"Result: {result}")
# print()

# Example 2: Data Pipeline
# class DataProcessor:
#     def __init__(self):
#         self.is-plain-object = is-plain-object_module
#
#     def process_batch(self, items):
#         """Process batch using is-plain-object"""
#         return [self.is-plain-object.default(item) for item in items]
#
# processor = DataProcessor()
# data = ['item1', 'item2', 'item3']
# results = processor.process_batch(data)
# for i, result in enumerate(results, 1):
#     print(f"  Item {i}: {result}")
# print()

print("Real-world use case:")
print("- Python data pipeline uses is-plain-object")
print("- Uses same TypeScript implementation as Node.js API")
print("- Guarantees consistent object checking across services")
print("- Validate data structures")
print()

print("Example: Cross-Service Architecture")
print("┌─────────────────────────────────────────┐")
print("│   Elide IsPlainObject (TypeScript)           │")
print("│   elide-is-plain-object.ts                        │")
print("└─────────────────────────────────────────┘")
print("         ↓                   ↓")
print("    ┌────────┐          ┌────────┐")
print("    │ Node.js│          │ Python │")
print("    │  API   │          │Pipeline│")
print("    └────────┘          └────────┘")
print("         ↓                   ↓")
print("    Same object checking everywhere!")
print()

print("Problem Solved:")
print("Before: Python + JavaScript = different object checking implementations")
print("After: One Elide implementation = 100% consistent behavior")
print()

print("When Elide Python API is ready, usage will be:")
print("  from elide import require")
print("  is-plain-object = require('./elide-is-plain-object.ts')")
print("  result = is-plain-object.default(data)")

File: conversions/is-plain-object/elide-is-plain-object.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-is-plain-object
#
# This demonstrates calling the TypeScript is-plain-object implementation from Ruby
# using Elide's polyglot capabilities.
#
# Benefits:
# - One is-plain-object implementation shared across TypeScript and Ruby
# - Consistent object checking behavior across microservices
# - No Ruby library discrepancies
# - Perfect for Rails, Sidekiq, background workers

puts "=== Ruby Calling TypeScript IsPlainObject ===\n"

# NOTE: Exact syntax depends on Elide's Ruby polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Ruby support is ready

# Assuming Elide provides something like:
# is-plain-object_module = Elide.require('./elide-is-plain-object.ts')

# Example 1: Basic Usage
# result = is-plain-object_module.default(input_data)
# puts "Result: #{result}"
# puts ""

# Example 2: Rails Integration
# class DataService
#   def initialize
#     @is-plain-object = is-plain-object_module
#   end
#
#   def process(data)
#     @is-plain-object.default(data)
#   end
# end
#
# service = DataService.new
# result = service.process(input_data)
# puts "Processed: #{result}"
# puts ""

# Example 3: Sidekiq Worker
# class ProcessingJob
#   include Sidekiq::Worker
#
#   def perform(data)
#     result = is-plain-object_module.default(data)
#     puts "Job result: #{result}"
#   end
# end

puts "Real-world use case:"
puts "- Ruby worker uses is-plain-object"
puts "- Uses same TypeScript implementation as Node.js API"
puts "- Guarantees consistent object checking across services"
puts "- Validate data structures"
puts ""

puts "Example: Microservices Architecture"
puts "┌─────────────────────────────────────────┐"
puts "│   Elide IsPlainObject (TypeScript)           │"
puts "│   elide-is-plain-object.ts                        │"
puts "└─────────────────────────────────────────┘"
puts "         ↓           ↓           ↓"
puts "    ┌────────┐  ┌────────┐  ┌────────┐"
puts "    │ Node.js│  │  Ruby  │  │ Rails  │"
puts "    │  API   │  │ Worker │  │  App   │"
puts "    └────────┘  └────────┘  └────────┘"
puts "         ↓           ↓           ↓"
puts "    Same object checking everywhere!"
puts ""

puts "Problem Solved:"
puts "Before: Ruby + JavaScript = different implementations"
puts "After: One Elide implementation = 100% consistent"
puts ""

puts "Rails Integration (when ready):"
puts "  # config/initializers/is-plain-object.rb"
puts "  IS_PLAIN_OBJECT = Elide.require('./elide-is-plain-object.ts')"
puts "  "
puts "  # app/services/data_service.rb"
puts "  class DataService"
puts "    def process(data)"
puts "      IS_PLAIN_OBJECT.default(data)"
puts "    end"
puts "  end"
